src/lib.rs

```diff
@@ -95,23 +106,33 @@ impl IRObject {
 
     pub fn get_metal_lib_binary(
         &self,
+        compiler: &IRCompiler,
         shader_stage: ffi::IRShaderStage,
-        dest_lib: &mut IRMetalLibBinary,
-    ) -> bool {
-        unsafe {
+    ) -> Result<IRMetalLibBinary, MetalLibNoBytecodeFoundError> {
+        let mtl_lib = IRMetalLibBinary::new(compiler);
+        if unsafe {
             self.funcs
-                .IRObjectGetMetalLibBinary(self.me, shader_stage, dest_lib.me)
+                .IRObjectGetMetalLibBinary(self.me, shader_stage, mtl_lib.me)
+        } {
+            Ok(mtl_lib)
+        } else {
+            Err(MetalLibNoBytecodeFoundError)
```

Comment:
Should we include the `shader_stage` as context? Maybe some info about `self.me` too?

Reply:
Added the object type and shader stage to the error code, which is fetched from the IRObject (self.me)